pandas/__init__.py

```diff
@@ -184,6 +147,89 @@
 del get_versions, v
 
 
+def __getattr__(key: str):
+    # lazy imports to speed up 'import pandas as pd'
+    if key == "eval":
+        from pandas.core.computation.api import eval
+
+        return eval
+    elif key == "testing":
+        import pandas.testing
+
+        return pandas.testing
+
+    elif key in {
+        "ExcelFile",
+        "ExcelWriter",
+        "read_excel",
+        "read_csv",
+        "read_fwf",
+        "read_table",
+        "read_pickle",
+        "to_pickle",
+        "HDFStore",
+        "read_hdf",
+        "read_sql",
+        "read_sql_query",
+        "read_sql_table",
+        "read_clipboard",
+        "read_parquet",
+        "read_orc",
+        "read_feather",
+        "read_gbq",
+        "read_html",
+        "read_xml",
+        "read_json",
+        "read_stata",
+        "read_sas",
+        "read_spss",
+    }:
+        import pandas.io.api
+
+        return getattr(pandas.io.api, key)
+    elif key == "json_normalize":
+        from pandas.io.json._normalize import json_normalize
+
+        return json_normalize
+    raise AttributeError(f"module 'pandas' has no attribute '{key}'")
+
+
+def __dir__() -> list[str]:
+    # include lazy imports defined in __getattr__ in dir()
+    base = list(globals().keys())
```

Comment:
Could this be simplified to `return __all__`?

Reply:
it could, but i think that would render the namespace tests redundant